### ISSUE TITLE
Image and Caption formatting in asciidoc💡

### ISSUE THREAD

**Issue by tiuweehan** [nus-cs2103-AY1920S1/forum#168]
Hi all, the following is a short guide on how to resize and position images and captions nicely in `asciidoc` for both HTML and PDF.

I couldn't find any resources only so it took a while to figure out how to format the images for my PPP. Hopefully this might be helpful for some of you =)

## Resizing Images

**Step 1.** The following is the standard way to embed images in `asciidoc`. By default, the width of the image in HTML will be 100% of the image size or 100% of the page width, whichever is smaller.
```
.This is a caption
image::my_image.png[]
```
translates to
![image](https://user-images.githubusercontent.com/39181981/68538267-bf7b2b80-03ac-11ea-8027-e27f90d49486.png)

**Step 2.** To resize the image for HTML, use the `width` attribute of the image. You can adjust the size using either the percentage of page width (e.g. `width=50%`) or by absolute size of the image (e.g. `width=10cm`).
```
.This is a caption
image::my_image.png[width=50%]
```
translates to
![image](https://user-images.githubusercontent.com/39181981/68538304-534cf780-03ad-11ea-910a-b4e8baebb849.png)

However, this might cause formatting issues when exporting to PDF, as seen below
![image](https://user-images.githubusercontent.com/39181981/68538750-8941aa00-03b4-11ea-8663-6f3c5d57d5cf.png)

**Step 3.** To adjust the width of an image in PDF, use the `scaledwidth` attribute instead. The `scaledwidth` attribute can be used together with the `width` attribute without causing formatting issues for HTMLl. However, do note that the `scaledwidth` attribute can only be adjusted by absolute size (e.g. `scaledwidth=12cm`)
```
.This is a caption
image::my_image.png[width=50%, scaledwidth=12cm]
```
which translates to
![image](https://user-images.githubusercontent.com/39181981/68538777-c1e18380-03b4-11ea-9748-c875d717202d.png)

## Positioning images
Images like UML diagrams and captions should ideally be aligned to center.  While it is possible to align images to the center in HTML, **it is currently not possible to do so for captions in HTML**. However, it is still possible to do so in PDF.

**Step 1.** To align images (but not captions) in the center for both HTML and PDF,. use the `align` attribute.
```
.This is a caption
image::my_image.png[align="center", width=50%, scaledwidth=12cm]
```
which translates to
![image](https://user-images.githubusercontent.com/39181981/68538462-c5bed700-03af-11ea-8bd5-8dc6ed91dd0e.png)

**Step 2.** To align both images AND captions in the center for PDF, a few more steps are needed. Instead of using the PDF feature in Intellij, we have to install a command line version of `asciidoctor-pdf` from [this link](https://asciidoctor.org/docs/asciidoctor-pdf/).

**Step 3.** By default, `asciidoc-pdf` aligns images and captions to the left, so we need to change this configuration. In the command line, execute the following command to find the directory where the ruby gems are stored
```
gem environment 
```
![image](https://user-images.githubusercontent.com/39181981/68538545-3ca89f80-03b1-11ea-8ca4-c12bf115219b.png)

In this case, my gems are stored at `/Users/tiuweehan/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0`

**Step 4.** Find the directory where `asciidoctor-pdf` is stored. In this case, mine is stored at `/(ruby gems directory)/gems/asciidoctor-pdf-1.5.0.beta.7/`.
![image](https://user-images.githubusercontent.com/39181981/68538604-3e269780-03b2-11ea-8db6-26a613f008bf.png)

**Step 5.** Edit the configuration file at `/(current directory)/data/theme/default-theme.yml`.
![image](https://user-images.githubusercontent.com/39181981/68538629-8c3b9b00-03b2-11ea-8d91-57963056c496.png)

**Step 6.** Change the value under `caption > align` from `left` to `center` (See line 132)
```
caption:
    align: center
```
![image](https://user-images.githubusercontent.com/39181981/68538638-d1f86380-03b2-11ea-9b28-368da5f8782c.png)

**Step 7.** Now cd to the directory where the adoc file is and use `asciidoctor-pdf` to export the adoc file to PDF.
```
asciidoctor-pdf my_ascii_doc.adoc -o my_ascii_doc.pdf
```
The exported PDF should now be aligned nicely
![image](https://user-images.githubusercontent.com/39181981/68538409-b8edb380-03ae-11ea-83fc-52e67f7af60f.png)

**Comment by geshuming** [comment]
For positioning, I modified the `asciidoctor.css` file in `/stylesheets` to align the imageblocks. 
[The commit I made can be found here](https://github.com/AY1920S1-CS2103T-W13-1/main/commit/1637f6f60f1423dd00a76585ec49ce15723ef91d) It was a small fix that didn't warrant installing applications/plugins, in my opinion.

**Comment by tiuweehan** [comment]
> For positioning in HTML, I modified the `asciidoctor.css` file in `/stylesheets` to align the imageblocks.
> [The commit I made can be found here](https://github.com/AY1920S1-CS2103T-W13-1/main/commit/1637f6f60f1423dd00a76585ec49ce15723ef91d) It was a small fix that didn't warrant installing applications/plugins, in my opinion.
> 
> > it is currently not possible to do so for captions in HTML
> 
> My commit also aligns the caption, so for anyone wondering if you can, you can.

This is even better!

**Comment by geshuming** [comment]
> Edit: Didn't manage to get this to work for me though :(

I don't think it shows up in the preview on the IDE, but it shows up for my site as well as deployment previews

**Comment by tiuweehan** [comment]
> > Edit: Didn't manage to get this to work for me though :(
> 
> I don't think it shows up in the preview on the IDE, but it shows up for my team's site as well as deployment previews

Oh you're right, I see it in my deployment previews now. But it doesn't seem to work for exporting to pdf through IntelliJ though
